clients/adapter-kafka/src/main/java/org/eclipse/hono/adapter/client/command/kafka/KafkaBasedInternalCommandConsumer.java

```diff
@@ -232,17 +236,30 @@ private String getTopicName() {
 
     void handleCommandMessage(final KafkaConsumerRecord<String, Buffer> record) {
 
+        // get partition/offset of the command record - related to the tenant-based topic the command was originally received in
+        final Integer commandPartition = KafkaRecordHelper
+                .getHeaderValue(record.headers(), KafkaRecordHelper.HEADER_ORIGINAL_PARTITION, Integer.class)
+                .orElse(null);
+        final Long commandOffset = KafkaRecordHelper
+                .getHeaderValue(record.headers(), KafkaRecordHelper.HEADER_ORIGINAL_OFFSET, Long.class)
+                .orElse(null);
+        if (commandPartition == null || commandOffset == null) {
+            LOG.info("command record is invalid - missing required original partition/offset headers");
+            return;
+        }
+
         final KafkaBasedCommand command;
         try {
             command = KafkaBasedCommand.fromRoutedCommandRecord(record);
         } catch (final IllegalArgumentException e) {
-            LOG.debug("command record is invalid [tenant-id: {}, device-id: {}]",
+            LOG.info("command record is invalid [tenant-id: {}, device-id: {}]",
```

Comment:
FMPOV this should be `LOG.warn()` as it indicates something went wrong with the command router.